packages/react/src/components/Loader/Loader.tsx

```diff
@@ -94,9 +97,40 @@ class Loader extends UIComponent<WithAsProp<LoaderProps>, LoaderState> {
 
     this.state = {
       visible: this.props.delay === 0,
+      labelLoaderId: getOrGenerateIdFromShorthand('loader-label-', props.label, ''),
     }
   }
 
+  static getIdFromShorthand = <P extends Record<string, any>>(
+    value: ShorthandValue<P>,
+  ): string | undefined => {
+    if (_.isNil(value)) {
+      return undefined
+    }
+    let result = ''
+    if (React.isValidElement(value)) {
+      result = (value as React.ReactElement<{ id?: string }>).props.id
+    } else if (_.isPlainObject(value)) {
+      result = (value as Record<string, any>).id
+    }
+
+    return result
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    if (Loader.getIdFromShorthand(props.label) !== Loader.getIdFromShorthand(state.labelLoaderId)) {
```

Comment:
Shouldn't we compare here `getIdFromShorthand(props.label)` with `getIdFromShorthand(this.props.label)`?